File: solutions/python/grains/1/grains.py
"""
Finds the exponent of base 2
"""
def square(number):
    """
    Finds the number of grains for a given position square on the chessboard
    """
    if 0 < number <= 64:
        return 2**(number - 1)

    raise ValueError("square must be between 1 and 64")

"""
Finds the total
"""
def total():
    """
    Finds the total grains in the chessboard
    """
    n = 0
    for i in range(1, 64+1):
        n += square(i)

    return n

File: solutions/c/grains/1/grains.c
#include "grains.h"
#include <stdio.h>

#define SQUARES 64

uint64_t square(uint8_t index)
{
    uint64_t square = 0;
    
    if(index > 0 || index <= 64){
        index--;
        square = 1ull;
        
        while (index > 0){
            square *= 2;
            index--;
        }
    }

    return square;
}

uint64_t total(void){
    uint64_t total = 0;
    uint8_t squares = SQUARES;
    
    while (squares) {
        total += square(squares);
        squares--;
    }
    return total;
}